### ISSUE TITLE
NVDA on Chrmoe does not announce li item within error container

### ISSUE THREAD

**Issue by Imhiralmehta** [nvaccess/nvda#13165]
### Steps to reproduce:
Any error container with `role=alert` which populate on form submission, and no. of errors as a list (`ul` or `ol`) inside it. Rough explanation here
```
Here is an error message

- error 1
- error 2
- error 3
```
Edited to include modified html snippet as well as speech viewer output
HTML: (this is just snipped of error message container, its a result of form submission)
```
<div class="error" type="danger" appearance="standard">
                <div class="message" role="alert">
				<svg>error icon</svg>
				<div class="message-content">
				<p>The following 3 fields have an error:</p>
                                      <ul>
                                              <li>error 1</li>
                                              <li>error 2</li>
                                              <li>error 3</li>
                                      </ul>
				</div>
		</div>
				
            </div>
```
Speech viewer output:
```
banner landmark    The page navigation is complete. You may now navigate the page content as you wish.
main landmark    Back to xyz  visited  link  
clickable  , abc  menu button  subMenu  required  
clickable  , def  menu button  subMenu  required  
ghi
jkl  radio button  not checked  1 of 3
mno  edit  required  selected pqr
clickable  , stu  menu button  subMenu  required  
vwx  button  
alert  Please fix the following 3 errors:
```

### Actual behavior:
NVDA default scan mode only read "Please fix the following 3 errors"
### Expected behavior:
NVDA should read everything - "Please fix the following 3 errors. List of 3 errors. error 1, error 2, error 3"
### System configuration
Tested on Chrome 96.0.4664.93
#### NVDA installed/portable/running from source:
Installed version
#### NVDA version:
2021.2
#### Windows version:
Windows 10
#### Other information about your system:
Tested by multiple people with same result. Edge with Narrator has no issue. Firefox with NVDA does not announce error container at all.
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not sure

**Comment by feerrenrut** [comment]
@Imhiralmehta Could you update the description with a `HTML` snippet or link to a service like CodePen, jsbin, jsFiddle.
Also please give the keys you pressed to test this and the speech output from NVDA, an easy way to do this is to first press `NVDA+4` to turn on reporting of command keys and enable the speech viewer (`NVDA+n` to open the NVDA menu, `t` for tools, `s` for speech viewer). Then you can reproduce the issue and copy from the speech viewer.

This will help to ensure that our testing of this matches yours, reducing the ambiguity of this bug report.

**Comment by Imhiralmehta** [comment]
@feerrenrut I have added html snippet and speech viewer output after modifying certain specific details.

**Comment by jeremyZX** [comment]
I'm experiencing this as well and I'm glad to see I'm not the only one, because I can't find any ARIA documentation on whether I'm doing something incorrectly.

This seems to happen when the alert is dynamically added to the DOM. If the alert is already present in the DOM before receiving content, then the items in the list are read. This is also modified by the role given to the element within the dynamically added alert; adding `role="none"` to the `<ul>` causes the list items to be read.

I created [a JSFiddle to demonstrate the effect](https://jsfiddle.net/a7g96hb5). The "Trigger Alert 1" button will announce both the textnode and the subsequent list. The "Trigger Alert 2" button only announces the textnode. Here's my output from the speech viewer. This is using Chrome 105 on Windows 10.

```
main landmark    frame    Trigger Alert 1  button    Alert 1 updates a static element with role="alert" present in the HTML document
Please correct the following issues:• One • Two   

Trigger Alert 2  button    Alert 2 dynamically injects an element with role="alert" into the HTML document
alert  Please correct the following issues:
```

**Comment by feerrenrut** [comment]
Thanks @jeremyZX 
We can reproduce this in Chrome, neither example works in firefox.

Superficially the example looks valid, we'll need to investigate to determine where the bug lies.

**Comment by Adriani90** [comment]
This is still reproducible both in Firefox and Chrome with NVDA 2024.4.1.

The ul element and its items are not included in the allert message.

I wonder whether the content of the allert is clearly specified in this code example?
Maybe @jcsteh has some suggestions here.

**Comment by CorgiDev** [comment]
Hi, I was looking to see if an issue similar to one I am experiencing had been reported and noticed another issue that may be related to this one. It is [#14990](https://github.com/nvaccess/nvda/issues/14990). Just wanted to add this note in case it might help.

**Comment by Adriani90** [comment]
@SaschaCowley is this a duplicate of #14990? They look quite similar to me.

**Comment by seanbudd** [comment]
Closing as a duplicate. #14990 is a cleaner minimal example.
